packages/plugin-dom-editable/src/DomEditable.ts

```diff
@@ -0,0 +1,202 @@
+     *
+     * @param action
+     */
+    private _matchCommand(action: NormalizedAction): [CommandIdentifier, object] {
+        switch (action.type) {
+            case 'insertLineBreak':
+                return ['insertLineBreak', {}];
+            case 'insertText':
+            case 'insertHtml': {
+                const params: InsertTextParams = { text: action.text };
+                return ['insertText', params];
+            }
+            case 'selectAll':
+                return ['selectAll', {}];
+            case 'setSelection': {
+                const vSelectionParams: VSelectionParams = {
+                    vSelection: this.dependencies
+                        .get(DomLayout)
+                        .parseSelection(action.domSelection),
+                };
+                return ['setSelection', vSelectionParams];
+            }
+            case 'insertParagraphBreak':
+                return ['insertParagraphBreak', {}];
+            case 'deleteWord':
+                // TODO: extract range
+                if (action.direction === Direction.FORWARD) {
+                    return ['deleteForward', { range: {} }];
+                } else {
+                    return ['deleteBackward', { range: {} }];
+                }
+            case 'deleteContent': {
+                if (action.direction === Direction.FORWARD) {
+                    return ['deleteForward', {}];
+                } else {
+                    return ['deleteBackward', {}];
+                }
+            }
+            default:
+                break;
+        }
+    }
+    async _onNormalizedEvent(batchPromise: Promise<EventBatch>): Promise<void> {
+        return this.editor.nextEventMutex(
+            async (): Promise<void> => {
+                const batch = await batchPromise;
+                let processed = false;
+                if (batch.inferredKeydownEvent) {
+                    processed = !!(await this._processKeydown(
+                        new KeyboardEvent('keydown', {
+                            ...batch.inferredKeydownEvent,
+                            key: batch.inferredKeydownEvent.key,
+                            code: batch.inferredKeydownEvent.code,
+                        }),
+                    ));
+                }
+                if (!processed) {
+                    for (const action of batch.actions) {
+                        const commandSpec = this._matchCommand(action);
+                        if (commandSpec) {
+                            await this.editor.execCommand(...commandSpec);
+                        }
+                    }
+                }
+            },
+        );
+    }
+    /**
+     * KeyboardEvent listener to be added to the DOM that calls `execCommand` if
+     * the keys pressed match one of the shortcut registered in the keymap.
+     *
+     * @param event
+     */
+    private async _processKeydown(event: KeyboardEvent): Promise<CommandIdentifier> {
```

Comment:
The logic to handle keydown should be on the layout manager as shortcuts could be done on any part of the layout.
However, this is true that some of the logic of the current function should stay here (the EventNormalizer).

Reply:
It's can be move on the DomLayout, and changed to use all interface.
But, when I tried to do, and it's strange to call the domLayout in '_onNormalizedEvent' method